### ISSUE TITLE
If multi "on" -> init will iterate many times

### ISSUE THREAD

**Issue by peterdenev** [krasimir/navigo#12]
Example:
```javascript
router.on('/', function(){ console.log('home') }
router.on('/about', function(){ console.log('about') }
router.on('/contacts', function(){ console.log('contacts') }
```
On init will log 3 times 'home'

[JSBin example](http://jsbin.com/figesifuqu/1/edit?js,console)

**Comment by krasimir** [comment]
That was a tricky one. When I started working on this project I wanted to escape from one thing - having an API method for the initial route resolving. Or in other words
```
add route
add route
add route
now try resolving the added routes
```
I design it like that
```
add route
now try resolving the added routes
add route
now try resolving the added routes
add route
now try resolving the added routes
```
I.e. after every `.on` the library calls `.resolve` method. The problem in your case could be solved by changing the order of the routes. I.e. you add the matching one at the end. As a rule which I missed adding to the documentation, I have "add the more specific rules first". That's why in your example you get three matches of `/`. It's because you added the matching route in the very beginning and then in every `.on` it successfully calls the handler.

Even though this could be solved by ordering the routes registration I decided to add a a third parameter to `.on` method which prevents the calling of `.resolve`. Or in other words:
```js
var router = new Navigo();

router.on('/', function () {
    console.log('/');   
}, true);
router.on('/about', function () {
    console.log('about');
}, true);
router.on('/contacts', function () {
    console.log('contacts');    
});
```
works as expected. You see only one `"/"` in the output. [jsbin](http://jsbin.com/xofolef/1/edit?js,console)
